### PR TITLE
MdePkg: Added definition of AMD specific public MSRs

### PR DESCRIPTION
**Pull request by VivianNK**
# Description

Added definition of AMD specific public MSRs:
1. SMBASE
2. SMM_ADDR
3. SMM_MASK

Replaced local Msr defines where applicable by including `Register/Amd/Msr.h`

---
Cherry-picked first commit from:
https://github.com/microsoft/mu_basecore/commit/2e2958c6d448fe8a31c71528ff954279c2bfeaf1

---

- [ ] Breaking change?
- [ ] Impacts security?
- [ ] Includes tests?

## How This Was Tested

Tested in Project MU

## Integration Instructions

N/A

### CODE REVIEW

**Review by mdkinney — COMMENTED** [review]
Will there be other changes to use these MSRs?  I would expect these are already used elsewhere in edk2 repo.  Can those changes be include in this same PR?

**Comment by VivianNK** [comment]
> Will there be other changes to use these MSRs? I would expect these are already used elsewhere in edk2 repo. Can those changes be include in this same PR?

Definitions for two of them can be found here as well:
https://github.com/tianocore/edk2/blob/bbee1cc852fa8676ed0b530b1c67c92f32f4f740/UefiCpuPkg/Library/SmmCpuFeaturesLib/AmdSmmCpuFeaturesLib.c#L24

@kuqin12 Should these older macros be replaced as a breaking change?
Do we want the new macros to be unsigned longs?

**Comment by kuqin12** [comment]
> > Will there be other changes to use these MSRs? I would expect these are already used elsewhere in edk2 repo. Can those changes be include in this same PR?
> 
> Definitions for two of them can be found here as well:
> 
> https://github.com/tianocore/edk2/blob/bbee1cc852fa8676ed0b530b1c67c92f32f4f740/UefiCpuPkg/Library/SmmCpuFeaturesLib/AmdSmmCpuFeaturesLib.c#L24
> 
> @kuqin12 Should these older macros be replaced as a breaking change? Do we want the new macros to be unsigned longs?

I think it does make sense to switch over the commonly defined macros. Given that each module has been using their own macro definition, the change should be very self-contained and thus not a breaking change.

**Review by changab — COMMENTED** [review]
This change looks to me good , but could you please move the Smm profile changes to a separate commit?
Thanks

File: UefiCpuPkg/PiSmmCpuDxeSmm/SmmProfile.c
Comment (VivianNK) [reply UefiCpuPkg/PiSmmCpuDxeSmm/SmmProfile.c]: Thanks, removed this include as I see it's getting the `Intel/ArchitecturalMsr.h` through `PiSmmCpuCommon.h` -> `Register/Intel/Msr.h`

**Review by changab — APPROVED** [review]

**Comment by changab** [comment]
@VivianNK please have UefiCpuPkg maintainers to merge this PR if there is no pending feedback needs to be addressed. Thanks.

**Comment by VivianNK** [comment]
@niruiyu @rahul1-kumar @kraxel @jiaxinwu 
Can I please get reviews on this?

**Review by kraxel — APPROVED** [review]

**Review by niruiyu — APPROVED** [review]

**Review by lgao4 — APPROVED** [review]

**Pull request merged**
